packages/mui-base/src/useTransitionStatus/useTransitionStatus.ts

```diff
@@ -0,0 +1,43 @@
+'use client';
+import * as React from 'react';
+import { useEnhancedEffect } from '../utils/useEnhancedEffect';
+import type { TransitionStatus } from './useTransitionStatus.types';
+
+/**
+ * Provides a status string for CSS transitions and animations for conditionally-rendered
+ * components.
+ * @param isRendered - a boolean that determines if the component is rendered.
+ * @ignore - internal hook.
+ */
+export function useTransitionStatus(isRendered: boolean) {
```

Comment:
How do you feel about returning props from this hook as well (`onTransitionEnd` and `onAnimationEnd`)? We may be able to omit `setMounted` from the returned object.

Reply:
I feel that should be an implementation detail of the components that use this hook instead. This keeps its responsibility quite minimal.